tests/indexes/arithmetic/float/test_floordiv.py

```diff
@@ -0,0 +1,131 @@
+    Never,
+    assert_type,
+)
+
+from tests import (
+    TYPE_CHECKING_INVALID_USAGE,
+    check,
+)
+
+
+@pytest.fixture
+def left() -> "pd.Index[float]":
+    """Left operand"""
+    lo = pd.Index([1.2, 2.4, 3.6])
+    return check(assert_type(lo, "pd.Index[float]"), pd.Index, np.floating)
+
+
+def test_floordiv_py_scalar(left: "pd.Index[float]") -> None:
+    """Test pd.Index[float] // Python native scalars"""
+    b, i, f, c = True, 1, 1.0, 1j
+    s, d = datetime(2025, 9, 27), timedelta(seconds=1)
+
+    check(assert_type(left // b, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(left // i, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(left // f, "pd.Index[float]"), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _03 = left // c  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _04 = left // s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _05 = left // d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    check(assert_type(b // left, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(i // left, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(f // left, "pd.Index[float]"), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _13 = c // left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _14 = s // left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+    check(assert_type(d // left, pd.TimedeltaIndex), pd.TimedeltaIndex, pd.Timedelta)
+
+
+def test_floordiv_py_sequence(left: "pd.Index[float]") -> None:
+    """Test pd.Index[float] // Python native sequences"""
+    b, i, f, c = [True, False, True], [2, 3, 5], [1.0, 2.0, 3.0], [1j, 1j, 4j]
+    s = [datetime(2025, 10, 27 + d) for d in range(3)]
+    d = [timedelta(seconds=s) for s in range(3)]
+
+    check(assert_type(left // b, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(left // i, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(left // f, "pd.Index[float]"), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _03 = left // c  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _04 = left // s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _05 = left // d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    check(assert_type(b // left, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(i // left, "pd.Index[float]"), pd.Index, np.floating)
+    check(assert_type(f // left, "pd.Index[float]"), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _13 = c // left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _14 = s // left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _15 = d // left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
```

Comment:
Bit surprised here that `timedelta // left` works but a `list[timedelta] // left` does not..  Shouldn't it have picked up `Index[float].__rfloordiv__(list[timedelta])` ?